commit 6bf098b436b0
Author: Gijs Kruitbosch <gijskruitbosch@gmail.com>
Date:   Mon Sep 25 17:24:26 2017 +0100

    Bug 1402896 - Make the url bar strip javascript even when preceded by control characters. r=mak, a=sledru
    
    MozReview-Commit-ID: 5ZO8n5lfvnl
    
    --HG--
    extra : source : 638e145f6bba437642d55f7b2baf5458df61419a
---
 browser/base/content/browser.js                                         | 2 +-
 .../content/test/urlbar/browser_removeUnsafeProtocolsFromURLBarPaste.js | 1 +
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git browser/base/content/browser.js browser/base/content/browser.js
index 8b0fb0276d19..b73ab2a3dd7e 100755
--- browser/base/content/browser.js
+++ browser/base/content/browser.js
@@ -6128,7 +6128,7 @@ function stripUnsafeProtocolOnPaste(pasteData) {
   // LOAD_FLAGS_DISALLOW_INHERIT_PRINCIPAL for those.
   let changed = false;
   let pasteDataNoJS = pasteData.replace(/\r?\n/g, "")
-                               .replace(/^(?:\s*javascript:)+/i,
+                               .replace(/^(?:\W*javascript:)+/i,
                                         () => {
                                                 changed = true;
                                                 return "";
diff --git browser/base/content/test/urlbar/browser_removeUnsafeProtocolsFromURLBarPaste.js browser/base/content/test/urlbar/browser_removeUnsafeProtocolsFromURLBarPaste.js
index 6f6682d51688..27129297b0a3 100644
--- browser/base/content/test/urlbar/browser_removeUnsafeProtocolsFromURLBarPaste.js
+++ browser/base/content/test/urlbar/browser_removeUnsafeProtocolsFromURLBarPaste.js
@@ -7,6 +7,7 @@ var pairs = [
   ["javascript:", ""],
   ["javascript:1+1", "1+1"],
   ["javascript:document.domain", "document.domain"],
+  [" \u0001\u0002\u0003\u0004\u0005\u0006\u0007\u0008\u0009javascript:document.domain", "document.domain"],
   ["java\nscript:foo", "foo"],
   ["http://\nexample.com", "http://example.com"],
   ["http://\nexample.com\n", "http://example.com"],
